### DISCUSSION TITLE
Lock multifactor authentication settings for a user

### DISCUSSION THREAD

**Discussion by tobias-urdin** [zitadel/zitadel#4547] [Q&A] [answered]
Hello 👋 

Super interesting project and thus far it has pretty much everything needed that I have been able to verify for our multiple use-cases. The one question I have, is it possible to lock multifactor authentication settings for a user?

We have a use-case where we enroll users ourselves (registration is disabled), give them a U2F hardware device but we don't want them to be able to remove it from their account or add/replace it with other devices/types of multifactor authenticators, is that possible?

Best regards

**Comment by fforootd** [comment]
Hi There

Great question!

At the moment we do not have a policy in place that can define what factors are allowed to be changed. I think there was a resent discussion on such things as preventing the users from changing there usernames.

@hifabienne do you no where that was discussed? I could not find it through the search.

**Reply by tobias-urdin** [reply]
Thanks!

**Comment by hifabienne** [comment, marked as answer]
Hi there. 

Yes we have this already in a issue.
https://github.com/zitadel/zitadel/issues/3965

This will probably need some more changes in ZITADEL since it needs some "permissions" on an object or attribute level. More detailed than we have right now with the Managers on level Instance, Organization and Projects

**Reply by tobias-urdin** [reply]
Thanks!
